client/chat/channel-info-card.tsx

```diff
@@ -0,0 +1,157 @@
+import { useStableCallback } from '../state-hooks'
+import { colorError } from '../styles/colors'
+import { body1, Body1, Headline6 } from '../styles/typography'
+import { getChannelInfo, joinChannel, navigateToChannel } from './action-creators'
+
+const Container = styled(Card)`
+  width: 100%;
+  max-width: 440px;
+  min-width: 220px;
+  padding: 16px;
+
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+`
+
+const StyledLoadingDotsArea = styled(LoadingDotsArea)`
+  width: 100%;
+`
+
+const StyledChannelIcon = styled.svg`
+  width: 56px;
+  height: auto;
+  flex-shrink: 0;
+`
+
+const ErrorChannelIcon = styled(StyledChannelIcon)`
+  color: ${colorError};
+`
+
+const ChannelInfoContainer = styled.div`
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+  margin: 0 8px;
+`
+
+const ErrorText = styled.div`
+  ${body1};
+  color: ${colorError};
+`
+
+export interface ConnectedChannelInfoCardProps {
+  channelId: SbChannelId
+  channelName: string
+}
+
+export function ConnectedChannelInfoCard({
+  channelId,
+  channelName,
+}: ConnectedChannelInfoCardProps) {
+  const dispatch = useAppDispatch()
+  const channelInfo = useAppSelector(s => s.chat.idToInfo.get(channelId))
+  const isUserInChannel = useAppSelector(s => s.chat.joinedChannels.has(channelId))
+
+  const [findChannelError, setFindChannelError] = useState<Error>()
+  const [joinChannelError, setJoinChannelError] = useState<Error>()
+  const [isJoinInProgress, setIsJoinInProgress] = useState(false)
+
+  useEffect(() => {
+    const abortController = new AbortController()
+
+    dispatch(
+      getChannelInfo(channelId, {
+        signal: abortController.signal,
+        onSuccess: () => setFindChannelError(undefined),
+        onError: err => setFindChannelError(err),
+      }),
+    )
+
+    return () => {
+      abortController.abort()
+    }
+  }, [channelId, dispatch])
+
+  const onViewClick = useStableCallback(() => {
+    navigateToChannel(channelId, channelName)
+  })
+
+  const onJoinClick = useStableCallback(() => {
+    setIsJoinInProgress(true)
+    dispatch(
+      joinChannel(channelName, {
+        onSuccess: channel => navigateToChannel(channel.id, channel.name),
+        onError: err => {
+          setJoinChannelError(err)
+          setIsJoinInProgress(false)
+        },
+      }),
+    )
+  })
+
+  const isChannelNotFound =
+    isFetchError(findChannelError) && findChannelError.code === ChatServiceErrorCode.ChannelNotFound
+  const isUserBanned =
+    isFetchError(joinChannelError) && joinChannelError.code === ChatServiceErrorCode.UserBanned
+
+  const icon =
+    channelInfo && (!channelInfo.private || isUserInChannel) ? (
+      <StyledChannelIcon as={ChannelIcon} />
+    ) : (
+      <ErrorChannelIcon as={WarningIcon} />
+    )
+
+  let subtitle
+  if (isChannelNotFound) {
+    subtitle = (
+      <ErrorText>
+        We couldn't find this channel, it might not exist or it has been re-created by someone else
```

Comment:
nit: This channel could not be found. It might not exist, or it may have been re-created by someone else.